src/metrics/representation.py

```diff
@@ -195,18 +195,22 @@ def prepare_representation_data(self, vectors: np.ndarray, queries_idxs: np.ndar
         
         relevant = []
         empty_relevant_idxs = []
+        # print(f'scores = {scores}')
         for idx in range(len(q_vecs)):
             relevant_idxs = np.where(scores[:, queries_idxs[idx]] > 0.)[0]
             if len(relevant_idxs) == 0:
                 empty_relevant_idxs.append(idx)
             else:
+                # Need sort relevant indexes by its scores for NDCG metric
+                current_scores = scores[relevant_idxs, queries_idxs[idx]]
+                sort_indexes = np.argsort(current_scores)
+                relevant_idxs = relevant_idxs[sort_indexes[::-1]]
```

Comment:
You sorted relevant indices but not scores

Reply:
Doesn't need to sort the scores, because scores is the marix with query-relevant information, but it's good to sort relevant indexes beforehand, so that for every query we have sorted by score its relevant indexes.